fault-proof/src/proposer.rs

```diff
@@ -211,6 +211,32 @@ where
 
         // Initialize state with anchor L2 block number
         let anchor_l2_block = factory.get_anchor_l2_block_number(config.game_type).await?;
+
+        // Validate contract configuration
+        let finalized_l2_block = host
+            .get_finalized_l2_block_number(&fetcher, anchor_l2_block.to::<u64>())
+            .await?
+            .ok_or_else(|| {
+                anyhow::anyhow!(
+                    "Cannot fetch finalized L2 block number from L2 RPC: {}\n\
+                     Please check that your L2 node is running and accessible.",
+                    config.l2_rpc
+                )
+            })?;
+
+        if anchor_l2_block > U256::from(finalized_l2_block) {
```

Comment:
Should we also prevent our config binaries from generating corrupted JSON when a user passes a `STARTING_BLOCK_NUMBER` that’s higher than the (finalized_l2_block_number -num_blocks_for_finality) ? Or at least emit a warning?

Reply:
```rust
let optimism_output_data: Value = data_fetcher
    .fetch_rpc_data_with_mode(
        RPCMode::L2Node,
        "optimism_outputAtBlock",
        vec![starting_block_number_hex.into()],
    )
    .await?;
```
I expect fetching output root for the config file to return error if it's such case. Do you think we should handle this proactively?

Reply:
I see. If it errors out in that case, don’t need then.